gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetTest.java

```diff
@@ -0,0 +1,323 @@
+    Mockito.when(srcIcebergTable.getDatasetDescriptor(Mockito.any())).thenReturn(Mockito.mock(DatasetDescriptor.class));
+    Mockito.when(destIcebergTable.getDatasetDescriptor(Mockito.any())).thenReturn(Mockito.mock(DatasetDescriptor.class));
+
+    icebergPartitionFilterPredicateUtil = Mockito.mockStatic(IcebergPartitionFilterPredicateUtil.class);
+    icebergPartitionFilterPredicateUtil
+        .when(() -> IcebergPartitionFilterPredicateUtil.getPartitionColumnIndex(Mockito.anyString(), Mockito.any(), Mockito.any()))
+        .thenReturn(Optional.of(0));
+
+    copyConfigProperties.setProperty("data.publisher.final.dir", "/test");
+    srcFilePaths = new ArrayList<>();
+  }
+
+  @AfterMethod
+  public void cleanUp() {
+    srcFilePaths.clear();
+    icebergPartitionFilterPredicateUtil.close();
+  }
+
+  private void setupSrcFileSystem() throws IOException {
+    sourceFs = Mockito.mock(FileSystem.class);
+    Mockito.when(sourceFs.getUri()).thenReturn(SRC_FS_URI);
+    Mockito.when(sourceFs.makeQualified(any(Path.class)))
+        .thenAnswer(invocation -> invocation.getArgument(0, Path.class).makeQualified(SRC_FS_URI, new Path("/")));
+    Mockito.when(sourceFs.getFileStatus(any(Path.class))).thenAnswer(invocation -> {
+      Path path = invocation.getArgument(0, Path.class);
+      Path qualifiedPath = sourceFs.makeQualified(path);
+      return getFileStatus(qualifiedPath);
+    });
+  }
+
+  private void setupDestFileSystem() throws IOException {
+    targetFs = Mockito.mock(FileSystem.class);
+    Mockito.when(targetFs.getUri()).thenReturn(DEST_FS_URI);
+    Mockito.when(targetFs.makeQualified(any(Path.class)))
+        .thenAnswer(invocation -> invocation.getArgument(0, Path.class).makeQualified(DEST_FS_URI, new Path("/")));
+    // Since we are adding UUID to the file name for every file while creating destination path,
+    // so return file not found exception if trying to find file status on destination file system
+    Mockito.when(targetFs.getFileStatus(any(Path.class))).thenThrow(new FileNotFoundException());
+  }
+
+  @Test
+  public void testGenerateCopyEntities() throws IOException {
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file1.orc");
+    List<DataFile> srcDataFiles = getDataFiles();
+    Mockito.when(srcIcebergTable.getPartitionSpecificDataFiles(Mockito.any())).thenReturn(srcDataFiles);
+
+    icebergPartitionDataset = new TestIcebergPartitionDataset(srcIcebergTable, destIcebergTable, properties, sourceFs,
+        true);
+
+    CopyConfiguration copyConfiguration =
+        CopyConfiguration.builder(targetFs, copyConfigProperties).preserve(PreserveAttributes.fromMnemonicString(""))
+            .copyContext(new CopyContext()).build();
+
+    Collection<CopyEntity> copyEntities = icebergPartitionDataset.generateCopyEntities(targetFs, copyConfiguration);
+
+    Assert.assertEquals(copyEntities.size(), 2);
+    verifyCopyEntities(copyEntities, true);
```

Comment:
the path of every copy entity needs validation.  pass that in the way we did in `IcebergDatasetTest::verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected)`

once you do, `copyEntities.size()` validation can and should be encapsulated within "verify"

Reply:
I am doing that but just in a different way - please have a look at function - 

```
private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, boolean sameSrcAndDestWriteLocation) {
    String srcWriteLocationStart = SRC_FS_URI + SRC_WRITE_LOCATION;
    String destWriteLocationStart = DEST_FS_URI + (sameSrcAndDestWriteLocation ? SRC_WRITE_LOCATION : DEST_WRITE_LOCATION);
    String srcErrorMsg = String.format("Source Location should start with %s", srcWriteLocationStart);
    String destErrorMsg = String.format("Destination Location should start with %s", destWriteLocationStart);
    for (CopyEntity copyEntity : copyEntities) {
      String json = copyEntity.toString();
      if (isCopyableFile(json)) {
        String originFilepath = getOriginFilePathAsStringFromJson(json);
        String destFilepath = getDestinationFilePathAsStringFromJson(json);
        Assert.assertTrue(originFilepath.startsWith(srcWriteLocationStart), srcErrorMsg);
        Assert.assertTrue(destFilepath.startsWith(destWriteLocationStart), destErrorMsg);
        String originFileName = originFilepath.substring(srcWriteLocationStart.length() + 1);
        String destFileName = destFilepath.substring(destWriteLocationStart.length() + 1);
        Assert.assertTrue(destFileName.endsWith(originFileName), "Incorrect file name in destination path");
        Assert.assertTrue(destFileName.length() > originFileName.length() + 1,
            "Destination file name should be longer than source file name as UUID is appended");
      } else{
        verifyPostPublishStep(json);
      }
    }
  }
```

Reply:
I understand what you're doing w/ the UUID - which *is* a good thing to validate - but the difference in your `verifyCopyEntities` method def is that it blindly verifies the length of the list without knowing specifically which paths should be there (i.e. it's missing a `List<String> expected` parameter).

why not take `List<String> expectedSrcFilePaths` to verify against each copyable file's `getOriginFilePathAsStringFromJson`?  then *in additional* continue to validate the relationship between each origin file path and its `getDestinationFilePathAsStringFromJson`.

Reply:
Done